File: sorting/sort.php
<?php

function quicksort($arr) {
    if (count($arr) <= 1) {
        return $arr;
    }
    
    $pivot = $arr[count($arr) / 2];
    $left = array();
    $middle = array();
    $right = array();
    
    foreach ($arr as $x) {
        if ($x < $pivot) {
            $left[] = $x;
        } elseif ($x > $pivot) {
            $right[] = $x;
        } else {
            $middle[] = $x;
        }
    }
    
    $left = quicksort($left);
    $right = quicksort($right);
    
    return array_merge($left, $middle, $right);
}

function main() {
    $size = 100000;  // Reduced for memory constraints
    $arr = array();
    
    mt_srand(42);
    for ($i = 0; $i < $size; $i++) {
        $arr[] = mt_rand(1, 1000000);
    }
    
    $sorted = quicksort($arr);
    
    echo "First 10: " . implode(" ", array_slice($sorted, 0, 10)) . "\n";
    echo "Last 10: " . implode(" ", array_slice($sorted, -10)) . "\n";
}

main();

?>

File: sorting/sort.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static List<int> QuickSort(List<int> arr)
    {
        if (arr.Count <= 1)
        {
            return arr;
        }
        
        int pivot = arr[arr.Count / 2];
        var left = new List<int>();
        var middle = new List<int>();
        var right = new List<int>();
        
        foreach (int x in arr)
        {
            if (x < pivot)
            {
                left.Add(x);
            }
            else if (x > pivot)
            {
                right.Add(x);
            }
            else
            {
                middle.Add(x);
            }
        }
        
        left = QuickSort(left);
        right = QuickSort(right);
        
        var result = new List<int>();
        result.AddRange(left);
        result.AddRange(middle);
        result.AddRange(right);
        
        return result;
    }
    
    static void Main()
    {
        int size = 1000000;
        var arr = new List<int>();
        
        Random rand = new Random(42);
        for (int i = 0; i < size; i++)
        {
            arr.Add(rand.Next(1, 1000001));
        }
        
        var sorted = QuickSort(arr);
        
        Console.Write("First 10: ");
        for (int i = 0; i < 10; i++)
        {
            Console.Write(sorted[i] + " ");
        }
        Console.WriteLine();
        
        Console.Write("Last 10: ");
        for (int i = size - 10; i < size; i++)
        {
            Console.Write(sorted[i] + " ");
        }
        Console.WriteLine();
    }
}
